### PR TITLE
Add pre_save signal.

### PR DESCRIPTION
**Pull request by decentral1se**
Closes https://github.com/mbi/django-rosetta/issues/191.

### All Submissions:

- [ ] Are tests passing? (From the root-level of the repository please run `pip install tox && tox`)
- [ ] I have added or updated a test to cover the changes proposed in this Pull Request
- [ ] I have updated the documentation to cover the changes proposed in this Pull Request

### CODE REVIEW

File: rosetta/views.py [hunk 1]
```diff
@@ -155,6 +155,8 @@ def _request_request(key, default=None):
                         storage.set('rosetta_last_save_error', True)
 
             if file_change and rosetta_i18n_write:
+                pre_save.send(sender=None, language_code=rosetta_i18n_lang_code, request=request)
```
Comment (mbi) [inline rosetta/views.py:158]: I wonder if the signal shouldn't be fired before this test? I.e. what is the semantics of the `pre_save` signal? Should it only be fired if we can actually save the PO file?

**Comment by mbi** [comment]
Thanks for this pull request, would you please consider including a test, maybe update the documentation?

File: rosetta/views.py (hunk 1 above)
Comment (mbi) [inline rosetta/views.py:158]: Also, I'd rather use `send_robust` instead of plain `send` here, or include some exception handling.

**Comment by decentral1se** [comment]
As it turns out, I don't need this. I kinda also don't have time.

Thanks for the review but I think I'll close this off.

Nice one for the work so far!

**Pull request closed without merging**